src/core/services/formly.config.ts

```diff
@@ -31,17 +32,23 @@ export class FormlyConfig {
     });
   }
 
-  setType(options: TypeOption) {
-    if (!this.types[options.name]) {
-      this.types[options.name] = <TypeOption>{};
-    }
-    this.types[options.name].component = options.component;
-    this.types[options.name].name = options.name;
-    this.types[options.name].extends = options.extends;
-    if (options.wrappers) {
-      options.wrappers.map((wrapper) => {
-        this.setTypeWrapper(options.name, wrapper);
+  setType(options: TypeOption | TypeOption[]) {
+    if (Array.isArray(options)) {
+      options.map(option => {
+        this.setType(option);
       });
+    } else if (this.formlyUtils.isObject(options)) {
```

Comment:
this test is useless here, if `options` is not an array then it's an instance of TypeOption
```ts
   ...
} else {
   if (!this.types[options.name]) {
   ...
}
```